edx_proctoring/views.py

```diff
@@ -386,6 +393,205 @@ def get(self, request):
+
+    **Exceptions**
+        HTTP GET:
+            * 404 if the requesting user is not staff or course staff for the course associated with
+            the supplied course ID
+            * 404 if there is no onboarding exam in the course associated with the supplied course ID
+    """
+    @method_decorator(require_course_or_global_staff)
+    def get(self, request, course_id):
+        """
+        HTTP GET handler.
+        """
+        # get last created onboarding exam if there are multiple
+        onboarding_exam = (ProctoredExam.get_practice_proctored_exams_for_course(course_id)
+                           .order_by('-created').first())
+        if not onboarding_exam or not get_backend_provider(name=onboarding_exam.backend).supports_onboarding:
+            return Response(
+                status=404,
+                data={'detail': _('There is no onboarding exam related to this course id.')}
+            )
+        serialized_onboarding_exam = ProctoredExamSerializer(onboarding_exam).data
+
+        data_page = request.GET.get('page', 1)
+        text_search = request.GET.get('text_search')
+        status_filter = request.GET.get('status')
+
+        enrollments = get_enrollments_for_course(course_id)
+        # filter down enrollments for users that can take proctored exams
+        allowed_enrollments_users = [
+            enrollment.user
+            for enrollment
+            in enrollments
+            if enrollment.user.has_perm('edx_proctoring.can_take_proctored_exam', serialized_onboarding_exam)
+        ]
+
+        # filter allowed_enrollments_users by text_search
+        allowed_enrollments_users = self._filter_users_by_username_or_email(allowed_enrollments_users, text_search)
+
+        # get exam attempts for users for the exam
+        exam_attempts = ProctoredExamStudentAttempt.objects.get_exam_attempts_for_users_by_exam_id(
+            onboarding_exam.id,
+            allowed_enrollments_users
+        ).values('user_id', 'status', 'modified')
+
+        # select the most recent exam attempt per user
+        exam_attempts_per_user = self._get_most_recent_exam_attempt_per_user(exam_attempts)
+
+        onboarding_data = []
+        for user in allowed_enrollments_users:
+            user_attempt = exam_attempts_per_user.get(user.id, {})
+
+            data = {}
+            data['username'] = user.username
+            data['status'] = (InstructorDashboardOnboardingAttemptStatus
+                              .get_onboarding_status_from_attempt_status(user_attempt.get('status')))
+            data['modified'] = user_attempt.get('modified')
+
+            onboarding_data.append(data)
+
+        # filter the data by status filter
+        # we filter late than the text_search because users without exam attempts
+        # will have an onboarding status of "not_started", and we want to be
+        # able to filter on that status
+        if status_filter:
```

Comment:
Do we want to be able to filter by multiple statuses here?

Reply:
Yes, thank you. I'll add that in.